src/sentry/utils/jwt.py

```diff
@@ -73,16 +79,15 @@ def decode(
         kwargs["audience"] = audience
     if algorithms is None:
         algorithms = ["HS256"]
-    # This type is checked in the tests so this is fine.
-    return pyjwt.decode(token, key, options=options, algorithms=algorithms, **kwargs)  # type: ignore
+    return pyjwt.decode(token, key, options=options, algorithms=algorithms, **kwargs)
 
 
 def encode(
-    payload: Mapping[str, str],
+    payload: JSONData,
     key: str,
     *,  # Force passing optional arguments by keyword
     algorithm: str = "HS256",
-    headers: Optional[Mapping[str, str]] = None,
+    headers: Optional[JSONData] = None,
```

Comment:
this is a more lax type than what was there before I think? Also I don't think http headers are json data... same for the `peek_header` func

Reply:
Yes it is more lax.  But it also needs to be, the JWT RFC explicitly says these are JSON objects and not all values are strings.  Same for `peek_header` indeed.

Reply:
oh, just realised your comment.  these are JWT headers, not HTTP headers.  So yes, it is explicitly a JSON object.